PrecacheGeolocation.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+if (isset($_SERVER['REQUEST_METHOD'])) {
+    die();
+} // Web clients die.
+
+ini_set('display_errors', 1);
+ini_set('memory_limit', '256M');
+
+require_once 'config.inc.php';
+require_once 'functions.php';
+require_once 'includes/PdoDatabase.php';
+
+/** @var PdoDatabase $database */
+$database = gGetDb();
+
+$locationProvider = new IpLocationProvider($database, $locationProviderApiKey);
+
+$done = false;
+while (!$done) {
+    echo "Beginning txn\n";
+    $database->beginTransaction();
+
+    try {
+        $statement = $database->query(<<<SQL
+            SELECT p.prox
+            FROM (
+              SELECT trim(substring_index(substring_index(r.forwardedip, ',', n.n), ',', -1)) prox
```

Comment:
**TL;DR:** It's splitting the `forwardedip` field from "a, b, c" to three rows: one each for `a`, `b`, and `c` using fancy substrings joined against a static list of numbers for distances.

Longer demo answer: https://phabricator.stwalkerster.co.uk/P137